llvm/docs/DeveloperPolicy.rst

```diff
@@ -336,12 +341,12 @@ you follow these guidelines to help review, search in logs, email formatting
 and so on. These guidelines are very similar to rules used by other open source
 projects.
 
-Most importantly, the contents of the message should be carefully written to
-convey the rationale of the change (without delving too much in detail). It
-also should avoid being vague or overly specific. For example, "bits were not
-set right" will leave the reviewer wondering about which bits, and why they
-weren't right, while "Correctly set overflow bits in TargetInfo" conveys almost
-all there is to the change.
+Commit messages should communicate briefly what the change does, and provide
+background, possibly with context, for why the change is being made. Commit
+messages should be thoughtfully written and specific, rather than vague. For
+example, "bits were not set right" will leave the reviewer wondering about which
+bits, and why they weren't right, while "Correctly set overflow bits in
+TargetInfo" conveys almost all there is to the change.
```

Comment:
I'd also remind that the final commit, potentially squashed from multiple commits in a PR, will have the description taken from the PR description. The latter is set once when the PR is created, defaulting to the commit message if the PR only contains one commit, and does _not_ track the changes to the commit message. As a corollary, developers must manually update the PR description as they iterate on the change...